src/hotspot/share/opto/predicates.cpp

```diff
@@ -130,10 +131,15 @@ bool RuntimePredicate::is_predicate(Node* node, Deoptimization::DeoptReason deop
   return RegularPredicateWithUCT::is_predicate(node, deopt_reason);
 }
 
+// An Assertion Predicate has always a true projection on the success path.
+bool AssertionPredicate::may_be_predicate_if(const Node* node) {
+  return node->is_IfTrue() && RegularPredicate::may_be_predicate_if(node->as_IfProj());
```

Comment:
Can you add `assert(node != nullptr` here in case this method will be used in other places?